### ISSUE TITLE
SECURITY BREACH using ?dir=

### ISSUE THREAD

**Issue by SeiRruf** [DirectoryLister/DirectoryLister#64]
I recently came across a security breach today while using this tool on a windows apache server.

You can use it to display any path OUTSIDE of the path the script runs in using the URL, if you know direct paths on the server. For instance, simply adding ?dir=C: to the end of the index.php will display the contents of the C:/ drive, and everything from there! This does not let the user display files in these directories, but seeing the directories themselves is already BAD enough!

Please be sure to patch this in future releases. It should *never* allow directory listing of parent directories.

----
Also, a RECOMMENDATION feature:
Would it be possible to run this script on a web-accessible directory, but have it browse files on a directory outside of the web directory? Say, I want to browse C:/My Pictures exclusively through this tool on domain.com/index.php (running your script). So in other words, your script would be a portal to make these files web-accessible.

Thanks!

**Comment by SeiRruf** [comment]
For those reading this, a quick patch is to add the following code right below <?php in your index.php file (assuming you use the default DirectoryLister PHP file)

ini_set ( 'open_basedir', dirname ( getcwd() ) );
if ( isset ( $_GET['dir'] ) && !@is_dir ( $_GET['dir'] ) ) {
	$_GET['dir'] = '.';
}

Which will only allow directories within the directory you allowed to be browsed. C:/, for example, will then default back to the root directory of which you expect the visitors to be browsing. This is not a perm fix, but it works until an update is released!

**Comment by PHLAK** [comment]
Thanks for finding this @SeiRruf.  I guess I haven't really tested this against a Windows server.  I'll be sure to get this fixed.

**Comment by SeiRruf** [comment]
No problem @PHLAK. :) Happy to help. I noticed it when trying to view a directory outside of my public_html, and noticed I was able to view the directories but not open files. This works with any absolute path, such as C: D: E: etc.

I'd love to see a feature where the path that is being listed is hidden through the script. For example, have the script in /home/ but have it listing files and directories from /files/images/ without giving away that path to the viewer at all. This would mean every single file and directory would have to be ran through this script. Images, etc. This would be useful for those who have a directory full of items they'd like to be accessible exclusively through this tool, without moving them to their public domain.

Thanks, and Happy New Year~
